src/domain/dex/mod.rs

```diff
@@ -146,7 +146,16 @@ impl Swap {
     }
 
     pub fn satisfies(&self, order: &domain::order::Order) -> bool {
-        self.output.amount * order.sell.amount >= self.input.amount * order.buy.amount
+        let output = self.output.amount.checked_mul(order.sell.amount);
```

Comment:
You can use `U256::full_mul` to implement this check without a possible overflow. If we only want to check that the prices are satisfied we don't need the multiplication to fit into a `U256`.